chain/consensus/iface.go

```diff
@@ -18,12 +18,35 @@ import (
 )
 
 type Consensus interface {
-	ValidateBlock(ctx context.Context, b *types.FullBlock) (err error)
+	// ValidateBlockHeader is called by peers when they receive a new block through the network.
+	//
+	// This is a fast sanity-check validation performed by the PubSub protocol before delivering
+	// it to the syncer. It checks that the block has the right format and it performs
+	// other consensus-specific light verifications like ensuring that the block is signed by
+	// a valid miner, or that it includes all the data required for a full verification.
 	ValidateBlockHeader(ctx context.Context, b *types.BlockHeader) (rejectReason string, err error)
+
+	// ValidateBlock is called by the syncer to determine if to accept a block or not.
+	//
+	// It performs all the checks needed by the syncer to accept
+	// the block (signature verifications, VRF checks, message validity, etc.)
```

Comment:
This is just a more heavy-weight version of `ValidateBlockHeader`, but I assume it is performed by the same nodes for the same purpose, and is just separated for performance optimization purposes. Please correct me if I'm wrong.

Reply:
Correct. The first one needs to be really light because it is done by the network protocol to prevent DDoS attacks (so it shouldn't be computationally expensive). `ValidateBlock` is the one you do to fully verify the block before executing it.

Reply:
Unfortunately, not. GossipSub is not the only way in which you can be receiving a new block. The syncer may also request blocks directly from known peers, in which case you will want to also do first the header (light) verification and then the full-block validation.